ldap/servers/slapd/back-ldbm/ldbm_modrdn.c

```diff
@@ -1557,6 +1566,200 @@ ldbm_back_modrdn(Slapi_PBlock *pb)
+    ID id,
+    IDList *children,
+    const Slapi_DN *dn_parentdn,
+    const Slapi_DN *dn_newsuperiordn,
+    struct backentry *child_entries[])
+{
+    /* Iterate over the children list renaming every child */
+    struct ldbminfo *li = (struct ldbminfo *)be->be_database->plg_private;
+    ldbm_instance *inst = (ldbm_instance *)be->be_instance_info;
+    struct backentry **child_entry_copies = NULL;
+    int retval = 0;
+    char **newsuperiordns = NULL;
+    size_t newsuperiordncomps = 0;
+    size_t parentdncomps = 0;
+    NIDS nids = children ? children->b_nids : 0;
+
+    /*
+     * Break down the parent entry dn into its components.
+     */
+    {
+        char **parentdns = slapi_ldap_explode_dn(slapi_sdn_get_dn(dn_parentdn), 0);
+        if (parentdns) {
+            for (; parentdns[parentdncomps] != NULL; parentdncomps++)
+                ;
+            slapi_ldap_value_free(parentdns);
+        } else {
+            return -1;
+        }
+    }
+
+    /*
+     * Break down the new superior entry dn into its components.
+     */
+    newsuperiordns = slapi_ldap_explode_dn(slapi_sdn_get_dn(dn_newsuperiordn), 0);
+    if (newsuperiordns) {
+        for (; newsuperiordns[newsuperiordncomps] != NULL; newsuperiordncomps++)
+            ;
+    } else {
+        return -1;
+    }
+
+    /*
+     * Iterate over the child entries renaming them.
+     */
+    child_entry_copies = (struct backentry **)slapi_ch_calloc(sizeof(struct backentry *), nids + 1);
+    for (size_t i = 0; i <= nids; i++) {
+        child_entry_copies[i] = backentry_dup(child_entries[i]);
+    }
+    for (size_t i = 0; retval == 0 && child_entries[i]; i++) {
+        retval = dsentrydn_moddn_rename_child(ptxn, be, li, child_entries[i], &child_entry_copies[i],
+                                              parentdncomps, newsuperiordns,
+                                              newsuperiordncomps);
+    }
+
+    if (0 == retval) {
+        for (size_t i = 0; child_entries[i]; i++) {
+            CACHE_REMOVE(&inst->inst_cache, child_entry_copies[i]);
+            CACHE_RETURN(&inst->inst_cache, &(child_entry_copies[i]));
+        }
+    } else {
+        /* failure */
+        for (size_t i = 0; child_entries[i]; i++) {
+            backentry_free(&(child_entry_copies[i]));
+        }
+    }
+
+    slapi_ch_free((void **)&child_entry_copies);
+    slapi_ldap_value_free(newsuperiordns);
+
+    return retval;
+}
+
+static int32_t
+dsentrydn_modrdn_update(
+    backend *be,
+    const char *olddn,
+    const char *newdn,
+    struct backentry *e,
+    struct backentry **ec,
+    ID id,
+    back_txn *txn)
+{
+    int32_t ret = 0;
+    int32_t cache_rc = 0;
+    ldbm_instance *inst = (ldbm_instance *)be->be_instance_info;
+
+    /* We have the entry, so update id2entry */
+    slapi_entry_attr_set_charptr((*ec)->ep_entry, SLAPI_ATTR_DS_ENTRYDN, newdn);
+    slapi_entry_set_dn((*ec)->ep_entry, (char *)newdn);
+    ret = id2entry_add_ext(be, *ec, txn, 1, &cache_rc);
```

Comment:
Is it updating the entryrdn index ? Should not entryrdn be updated ?

Reply:
No, the dsentry* functions are only meant to work on the "dsEntryDN" attribute (not entryrdn).  All the entryrdn functions are still called.